File: module_transition/src/main/java/kt/pdog18/com/module_transition/AutoFragment.java
package kt.pdog18.com.module_transition;

import android.support.transition.TransitionManager;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.OnClick;
import kt.pdog18.com.base.BaseFragment;


public class AutoFragment extends BaseFragment {

    @BindView(R.id.transitions_container)
    ViewGroup transitions_container;
    @BindView(R.id.text)
    TextView textView;
    private boolean visible;

    @Override
    protected int getLayoutId() {
        return R.layout.fragment_auto;
    }


    @OnClick(R.id.button)
    void button() {
        TransitionManager.beginDelayedTransition(transitions_container);
        textView.setVisibility((visible = !visible) ? View.VISIBLE : View.GONE);
    }
}

File: module_transition/src/main/java/kt/pdog18/com/module_transition/AutoFragment.kt
package kt.pdog18.com.module_transition

import android.os.Bundle
import android.support.transition.TransitionManager
import android.view.View
import kotlinx.android.synthetic.main.fragment_auto.*

import kt.pdog18.com.base.BaseFragment


class AutoFragment : BaseFragment() {

    private var visible: Boolean = false

    override fun getLayoutId(): Int {
        return R.layout.fragment_auto
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        button.setOnClickListener {
            TransitionManager.beginDelayedTransition(transitions_container)
            text.visibility = if (visible) View.VISIBLE else View.GONE
        }
    }
}
